source/code/plugins/in_npmd_server.rb

```diff
@@ -334,6 +343,34 @@ def check_and_update_binaries
             File.unlink(_x64BinPath) if File.exist?(_x64BinPath)
         end
 
+        def check_agent_capability
+            _deleteVersionFiles = false
+
+            if !File.exist?(@location_agent_binary)
+                _deleteVersionFiles = true
+            else
+                _getCapResult = `getcap #{@location_agent_binary}`
+                if !_getCapResult.include?(NPMD_AGENT_CAPABILITY)
+                    File.unlink(@location_agent_binary)
+                    _deleteVersionFiles = true
+                end
+            end
+
+            if _deleteVersionFiles
+                # Single/First workspace scenario
+                _fileLocPostFix = "#{NPMD_VERSION_FILE_DIR}/#{NPMD_VERSION_FILE_NAME}"
+                _loc = "#{@npmd_version_dir_prefix}/#{_fileLocPostFix}"
+                File.unlink(_loc) if File.exist?(_loc)
+                # Multiple workspace scenario
+                _stateDirs = Dir.glob("#{@npmd_version_dir_prefix}/**/#{NPMD_VERSION_FILE_DIR}")
```

Comment:
Have you tested this plugin on a machine with multiple workspaces onboarded, and that all version files get deleted?